src/effects/HalftoneEffect.ts

```diff
@@ -0,0 +1,160 @@
+import { HalftoneShape } from "../enums/HalftoneShape.js";
+import { Effect } from "./Effect.js";
+import { OverlayBlendFunction } from "./blending/blend-functions/OverlayBlendFunction.js";
+
+import fragmentShader from "./shaders/halftone.frag";
+import { Uniform } from "three";
+
+/**
+ * HalftoneEffect options.
+ *
+ * @category Effects
+ */
+
+export interface HalftoneEffectOptions {
+	shape?: HalftoneShape;
+	radius?: number;
+	rotateR?: number;
+	rotateG?: number;
+	rotateB?: number;
+	scatter?: number;
+}
+
+/**
+ * A halftone effect.
+ *
+ * Based on the implementation found in Three.js:
+ * https://github.com/mrdoob/three.js/blob/0bf3908b73b2cf73d7361cce17cfc8b816cb2a00/examples/jsm/postprocessing/HalftonePass.js
+ *
+ * @category Effects
+ */
+
+export class HalftoneEffect extends Effect {
+
+	/**
+	 * Constructs a new halftone effect.
+	 *
+	 * @param options - The options.
+	 */
+
+	constructor({
+		shape = HalftoneShape.ELLIPSE,
+		radius = 10,
+		rotateR = 14,
+		rotateG = 45,
+		rotateB = 30,
+		scatter = 0
+	}: HalftoneEffectOptions = {}) {
+
+		super("HalftoneEffectOptions");
+
+		this.fragmentShader = fragmentShader;
+		this.blendMode.blendFunction = new OverlayBlendFunction();
+
+		const uniforms = this.input.uniforms;
+		uniforms.set("radius", new Uniform(radius));
+		uniforms.set("rotateR", new Uniform(rotateR));
+		uniforms.set("rotateG", new Uniform(rotateG));
+		uniforms.set("rotateB", new Uniform(rotateB));
+		uniforms.set("scatter", new Uniform(scatter));
+		uniforms.set("shape", new Uniform(shape));
```

Comment:
Let's combine `rotationRGB` into one `vec3` uniform. Although WebGL performs [packing](https://www.khronos.org/webgl/public-mailing-list/public_webgl/1003/pdfG6k3c9fNgv.pdf), it's still safer to keep the amount of uniforms low. `Radius` and `scatterFactor` don't need to be combined.